### ISSUE TITLE
Repeater Injections for multiple item layouts

### ISSUE THREAD

**Issue by ChrisGV04** [soflyy/oxygen-bugs-and-features#1926]
One of the features I've really loved of JetEngine by Crocoblock is their ***"Listing Grid Injections"*** module, which allows us to set different layouts for the items inside of the ***repeater widget.*** In that way, we can set a different style for items by n number of appearance or for items that have a specific custom meta field value (dynamic conditions).

This would be very useful for listing ***items in the repeater that we may want to "stand out"*** from the rest based on meta fields, create ***dynamic layouts*** for listings, visually differentiating items  from certain categories, or simply ***add life to the list*** by having multiple attractive layouts for different items.

I hope I explain myself well enough. It would be wonderful to see this feature soon, I'd use it a lot and I think it could give life to more complex listings. You can check out the examples at Crocoblock's JetEngine Listing Grid Injections.

***Thank you for your time and for your amazing tool! Keep up the good work.***

**Comment by zuroma999** [comment]
Yes! Similarly, we'd love to replace Toolset Views with Oxygen repeaters. In Views, apart from allowing unrestricted custom code in the loop, they also expose the current loop index (0...n). This allows us to do things like add a call to action after the third item in a list, etc.

If Oxygen's condition interface had a condition to test the loop index in a repeater, that would go a long way to solving this. Or, through the conditions API, have a global variable that holds the loop index.
